pilot/pkg/xds/gen.go

```diff
@@ -79,8 +80,12 @@ func (s *DiscoveryServer) findGenerator(typeURL string, con *Connection) model.X
 	// some types to use custom generators - for example EDS.
 	g := con.proxy.XdsResourceGenerator
 	if g == nil {
-		// TODO move this to just directly using the resource TypeUrl
-		g = s.Generators["api"] // default to "MCP" generators - any type supported by store
+		if strings.HasPrefix(typeURL, "istio.io/debug/") {
+			g = s.Generators["event"]
```

Comment:
One issue - when we send a debug request, we register a watch for that type. This watch is never removed. As a result, all config changes will re-push a new debug response. Example logs:

```
2021-04-05T21:06:24.639273Z     info    ads     Push debounce stable[10] 1: 101.013956ms since last change, 101.013815ms since last push, full=true
2021-04-05T21:06:24.639415Z     info    ads     XDS: Pushing:2021-04-05T21:06:24Z/8 Services:5 ConnectedEndpoints:2  Version:2021-04-05T21:06:24Z/8
2021-04-05T21:06:24.640212Z     info    ads     CDS: PUSH for node:echo-cb96f8d94-ssbm8.default resources:24 size:17.1kB
2021-04-05T21:06:24.640247Z     info    ads     EDS: PUSH for node:echo-cb96f8d94-ssbm8.default resources:6 size:1.2kB empty:0 cached:6/6
2021-04-05T21:06:24.641634Z     info    ads     LDS: PUSH for node:echo-cb96f8d94-ssbm8.default resources:16 size:64.8kB
2021-04-05T21:06:24.641921Z     info    ads     RDS: PUSH for node:echo-cb96f8d94-ssbm8.default resources:6 size:4.3kB
2021-04-05T21:06:24.641971Z     info    ads     istio.io/debug/edsz: PUSH for node:echo-cb96f8d94-ssbm8.default resources:0 size:0B
2021-04-05T21:06:24.642000Z     info    ads     istio.io/debug/configz: PUSH for node:echo-cb96f8d94-ssbm8.default resources:0 size:0B
2021-04-05T21:06:24.642010Z     info    ads     istio.io/debug/config_dump with 0 ResourceNames
2021-04-05T21:06:24.642028Z     info    ads     istio.io/debug/config_dump: PUSH for node:echo-cb96f8d94-ssbm8.default resources:0 size:0B
2021-04-05T21:06:24.642075Z     info    ads     istio.io/debug/syncz: PUSH for node:echo-cb96f8d94-ssbm8.default resources:2 size:134B
2021-04-05T21:06:24.656383Z     info    ads     EDS: PUSH request for node:echo-cb96f8d94-ssbm8.default resources:15 size:3.0kB empty:0 cached:15/15

```